### ISSUE TITLE
Filter Analytics data for Top Earning Content/Top Earning Pages to only include revenue from AdSense

### ISSUE THREAD

**Issue by tofumatt** [google/site-kit-wp#8281]
## Feature Description

Currently, the Top Earning Content Key Metric Tile (#6248) and the Top Earning Pages widget with GA4 data (#8059) do _not_ filter out non-AdSense revenue. As discussed with @marrrmarrr in Slack (https://10up.slack.com/archives/CFFRMC5DE/p1708363912766809?thread_ts=1708361122.867059&cid=CFFRMC5DE), both these components' reports should filter out any data that does not include revenue sourced from AdSense.

The report data can be [filtered by AdSense `accountID`](https://10up.slack.com/archives/CBKKQEBR9/p1708348963677589?thread_ts=1708348556.944389&cid=CBKKQEBR9) to ensure only AdSense data is shown in Site Kit.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* Report data from Top Earning Content Key Metric Tile (#6248) and the Top Earning Pages widget with GA4 data (#8059) should only include revenue from the user's connected/linked AdSense account, and not from any other sources.

## Implementation Brief

* [x] In `assets/js/modules/adsense/components/dashboard/DashboardTopEarningPagesWidgetGA4.js` and `assets/js/modules/adsense/components/widgets/TopEarningContentWidget.js`:
    * [x] Get the `adsenseAccountID` using the `getAccountID` selector of AdSense module. 
    * [x] In the report arg:
        * [x] Add `adSourceName` to the list of dimensions. 
        * [x] Add `dimensionFilter` with the following Object to filter results by the `adsenseAccountID`
```js
{
  filter: {
    fieldName: 'adSourceName',
    stringFilter: {
      matchType: 'EXACT',
      value: `Google AdSense account (${ adsenseAccountID })`,
    },
  },
}
```
* [x]  In `includes/Modules/AdSense/Settings.php`:
    * [x] Implement `Setting_With_ViewOnly_Keys_Interface` to the `Setting` class to expose `accountID` in view-only context. 

### Test Coverage

* Update stories for both Widget to continue functioning properly. 

## QA Brief

* [ ] Enable the `ga4AdSenseIntegration` feature flag
* [ ] Simulate linking your AdSense and Analytics account by updating the `adSenseLinked` setting in the store (this will allow you to QA this ticket before #8049 is merged).
*You can achieve this by running the following the developer console:*
```
googlesitekit.data.dispatch('modules/analytics-4').setAdSenseLinked(true)
```
* [ ] Enable key metric widgets, and enable the "Top earning content" widget. Results from your site, if available, should appear in the widget.
* [ ] Earnings should exclude earnings from sources other than the AdSense account ID connected to SiteKit.
* [ ] The top earning pages key metric tile and dashboard widget should be viewable on the View Only dashboard if Adsense and Analytics modules are shared and the `adSenseLinked` setting is true (this can be simulated in view only mode as above).

When the `ga4AdSenseIntegration` feature flag is disabled, this KMW tile should not appear.

## Changelog entry

* Update Top Earning Content/Pages widgets to only show ad revenue from AdSense.

**Comment by benbowler** [comment]
Hey @mohitwp, thanks for your QA. I've created a new [PR](https://github.com/google/site-kit-wp/pull/8343) and put this back to CR to confirm the approach. Both widgets for me now show the same revenue values for each page.

To your second point, it is not possible to filter by adSourceName in the GA4 dashboard and there is no page data in the AdSense reporting, therefore the report URL we are using in the widget is the best link we can get.

**Comment by kuasha420** [comment]
@mohitwp Back to you for another pass.

**Comment by mohitwp** [comment]
### QA Update ⚠️ 

@benbowler The 'Top earning pages' KM widget and Dashboard widgets are displaying the same data for the first page in the list on the Main Dashboard. However, when we visit the related [report](https://analytics.google.com/analytics/web/?pli=1#/p285794360/reports/explorer?params=_r..nav%3Dmaui%26_u.date00%3D20240207%26_u.date01%3D20240305%26_r..dataFilters%3D%5B%7B%22type%22:1,%22fieldName%22:%22unifiedPagePathScreen%22,%22evaluationType%22:1,%22expressionList%22:%5B%22%252Fen%252Fwhat-do-l-or-n-car-stickers-mean-in-ireland%252F%22%5D,%22complement%22:false,%22isCaseSensitive%22:true,%22expression%22:%22%22%7D%5D&r=all-pages-and-screens) it shows different earnings.This discrepancy only occurs with the first page in the list; other pages display consistent earnings numbers on both the main dashboard and in the Analytics report.

<details>

![image](https://github.com/google/site-kit-wp/assets/94359491/6aaeea73-1332-485a-95a9-6abcf82e0e0d)

![image](https://github.com/google/site-kit-wp/assets/94359491/9c52084c-2026-4b1e-a6f0-3b2bafb80417)

![image](https://github.com/google/site-kit-wp/assets/94359491/51a81f3b-8e29-47c6-9f28-f69a63a2bbe4)


</details>

**Comment by benbowler** [comment]
Hey @mohitwp this is expected, unfortunately there is not a report in GA4 that we can link to that shows just the AdSense revenue (It's actually not possible in the GA4 interface to filter by the Ad Source Name as we can do in the API).

**Comment by mohitwp** [comment]
@benbowler Just wondering why it is happening only in case of first page in the list? Rest of the pages are showing same earning on main dashboard and under GA4.

**Comment by benbowler** [comment]
Hey @mohitwp, as I mentioned in Slack yesterday, by looking at some revenue reports in the GA dashboard I can confirm the only revenue source on oi.ie is the sites AdSense account.

I took another look at the widgets code an found that the `pageTitle` dimension, when removed, gives the correct full revenue amount for all pages. My hypothesis is the page title changed during the last 28 days meaning this dimension split the revenue shown in the widgets.

I've now created a [follow up PR](https://github.com/google/site-kit-wp/pull/8360) to bring the report numbers back into check, to match the report in GA.
